Sources/Purchasing/Purchases/TransactionPoster.swift

```diff
@@ -28,6 +28,7 @@ struct PurchasedTransactionData {
     var presentedOfferingContext: PresentedOfferingContext?
     var presentedPaywall: PaywallEvent?
     var unsyncedAttributes: SubscriberAttribute.Dictionary?
+    var metadata: [String: String]?
```

Comment:
I've been wondering whether it would make sense to make this a `[String: Any]` or something like that... but that might increase the scope quite a bit, so I think this is ok for now.